R/linelist_merge.R

```diff
@@ -0,0 +1,416 @@
+
+    datesnumeric<-output_sheet %>% select(contains("Date", ignore.case = T),id) %>% mutate_all(.funs=gsub,pattern=".",replacement = "/", fixed=T, perl = T) %>%
+      mutate_at(vars(contains("Date", ignore.case = T)), as.numeric) %>%
+      mutate_at(vars(contains("Date", ignore.case = T)),as.Date, origin="1899-12-30") %>%
+      mutate_at(vars(contains("Date", ignore.case = T)), as.character)
+
+      #handle dates that are characters. Select first 10 characters in cell This will select all full dates and not any times appended.
+      # Convert character formats using parse date time.
+      # Convert to numeric and change date format using origin of excel
+      # Convert to character date for final merge
+      datescharacter<-output_sheet %>% select(contains("Date", ignore.case = T), id) %>%
+        mutate_at(vars(contains("Date", ignore.case = T)), lubridate::parse_date_time, orders=c("ymd","Ymd","dmy","dmY","%Y-%m-%d","%y-%m-%d","%d-%m-%y","dBY","ymd HMS","Ymd HMS")) %>%
+        mutate_at(vars(contains("Date", ignore.case = T)),as.Date, origin="1899-12-30") %>% mutate_at(vars(contains("Date", ignore.case = T)),as.character)
+
+
+       #Natural join of data frames, fills in missing values in one df with values from other if not misssing
+
+      dates<- rquery::natural_join(datesnumeric, datescharacter,
+                                   by = "id",
+                                   jointype = "FULL")
+
+      # final date handling: change all columns to date class type (double)
+      dates_final<- dates %>% mutate_at(vars(contains("Date", ignore.case = T)),lubridate::parse_date_time, orders=c("ymd"))
+      # merge in these new handled dates to orginal sheet, removing old coded dates
+      date_vars<-dates_final %>% select(contains("Date", ignore.case = T)) %>% names(.)
+      output_sheet<-output_sheet[ , -which(names(output_sheet) %in% date_vars)]
+      #output_sheet$id<-rownames(output_sheet)
+      output_sheet<- merge(output_sheet,dates_final, by="id") %>% select(-c(id))
+
+
+
+      #remove rows where all columns except ID is missing or if case id is missing
+      output_sheet$na_rows <- rowSums(!is.na(output_sheet))
+      output_sheet<-  output_sheet %>% filter(!is.na(patinfo_id))  %>%  filter(na_rows > 1) %>% select(-c(na_rows))
+
+
+      # make column for ISO code
+      output_sheet$country_iso<-iso
+      #make variable names lower case for output sheet
+      names(output_sheet)<-tolower(names(output_sheet))
+
+  ######
+    #checking that variables of interest exists
+    #if not create missing column in cleaning we can use suurogate varaiables to infill
+
+    vars<- colnames(output_sheet)
+    output_sheet$id<-rownames(output_sheet)
+
+     # some countries missing this can use lab result date to infill in next clean?
+    if(!("report_date" %in% vars)){
+      output_sheet$report_date<-NA
+      output_sheet$report_date<-as.Date(output_sheet$report_date,origin="1899-12-30")
+      print("report_date")
```

Comment:
warnings... do a big combined message with all the added/dropped vars, e.g. [here](https://github.com/R4IDSR/covidmonitor/pull/9/files#diff-5581348288db36d823a2992aea1e71eb9b0847865fa1e4668c905c5a91734fbdR287)

Reply:
as above